bindings/matrix-sdk-ffi/src/sliding_sync.rs

```diff
@@ -26,29 +26,46 @@ use crate::{
     TimelineListener,
 };
 
+type StoppableSpawnCallback = Box<dyn FnOnce() + Send + Sync>;
+
 pub struct StoppableSpawn {
-    handle: Arc<RwLock<Option<JoinHandle<()>>>>,
+    handle: Arc<JoinHandle<()>>,
+    callback: Arc<RwLock<Option<StoppableSpawnCallback>>>,
```

Comment:
The `StoppableSpawn` type itself is never cloned AFAICT (it's put behind `Arc`), so why are its fields wrapped in `Arc`s?